pyvrp/cpp/crossover/crossover.h

```diff
@@ -19,26 +19,36 @@ void greedyRepair(std::vector<std::vector<int>> &routes,
                   ProblemData const &data);
 }  // namespace crossover
 
-typedef std::function<Individual(
-    std::pair<Individual const *, Individual const *> const &,
-    ProblemData const &,
-    PenaltyManager &,
-    XorShift128 &)>
-    CrossoverOperator;
-
 /**
- * Performs two SREX crossovers of the given parents (binary tournament). This
- * was one of ORTEC's DIMACS contributions.
+ * Performs two SREX crossovers of the given parents. SREX is a method that
+ * selects a set of routes for each parent and replaces the selected routes of
+ * the first parent with those of the second parent. The routes are selected by
+ * minimizing the overlap between the two sets of routes. This is achieved
+ * through a heuristic that iteratively shifts adjacent routes until no further
+ * improvement in minimizing the overlap is observed. Then, two offspring are
+ * generated by replacing the selected routes in two distinct ways, and the
+ * offspring with the lowest cost is returned.
+ *
+ * @param parents        The parent individuals.
+ * @param data           The problem data.
+ * @param penaltyManager The penalty manager.
+ * @param startA         Start index of adjacent routes in the first parent.
+ * @param startB         Start index of adjacent routes in the second parent.
+ * @param nMovedRoutes   Number of routes to move.
+ * @return The produced offspring.
+ *
  * <br />
- * Yuichi Nagata and Shigenobu Kobayashi. "A memetic algorithm for the pickup
- * and delivery problem with time windows using selective route exchange
- * crossover". In: International Conference on Parallel Problem Solving from
- * Nature. Springer. 2010, pp. 536–545.
+ * Yuichi Nagata and Shigenobu Kobayashi. "A memetic algorithm for
+ * the pickup and delivery problem with time windows using selective route
+ * exchange crossover". In: International Conference on Parallel Problem Solving
+ * from Nature. Springer. 2010, pp. 536–545.
  */
 Individual selectiveRouteExchange(
     std::pair<Individual const *, Individual const *> const &parents,
     ProblemData const &data,
     PenaltyManager const &penaltyManager,
-    XorShift128 &rng);
+    size_t startA,
+    size_t startB,
+    size_t const nMovedRoutes);
```

Comment:
The `A`, `B` terminology follows from Nagata and Kobayashi, I think. I vaguely recall reading about it when looking at edge assembly. I get that we use that in the implementation to match the paper, but in the interface can we maybe use a tuple like with the parents? Something like
```cpp
std::pair<size_t, size_t> starts,
```
or similar. Then our use of first/second matches that of the parents, and we don't need to use A/B outside of the SREX implementation (which I don't think we do anywhere else - we call things first/second in multiple places, but not A/B).

For `nMovedRoutes`, maybe `numMovedRoutes`?